### ISSUE TITLE
-ignore: help text says it matches error messages, implementation matches the rule name

### ISSUE THREAD

**Issue by on-keyday** [ultra-supara/sisakulint#591]
## Summary

The `-ignore` flag's help text says it matches **error messages**, but the implementation matches the **rule name**. A regular expression that matches the message text does not suppress anything, and an anchored rule name — which the help text implies would not match a message — suppresses everything from that rule.

Affected version: `v0.3.6` (tag resolves to `dea1408b296a54edc30206ee844daf5ea185b79e`).

## Current help text

```
  -ignore value
    	Regular expression matching to error messages you want to ignore. This flag is repeatable
```

## Measured behaviour

Workflow under test (`.github/workflows/w.yml`):

```yaml
name: T
on:
  pull_request_target:
  push:
    branches: [main]
jobs:
  j:
    runs-on: ubuntu-latest
    env:
      BAD NAME: x
    steps:
      - uses: actions/checkout@v4
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      - uses: actions/upload-artifact@v3
        with:
          path: .
      - run: echo "${{ github.event.pull_request.title }}"
```

Counting `[artipacked]` findings after `-ignore <pattern>`:

| `-ignore` pattern | what it looks like | findings left |
|---|---|---|
| `^artipacked$` | anchored rule name | 0 (suppressed) |
| `artipacked` | rule name substring | 0 (suppressed) |
| `\[artipacked\]` | how the rule appears in output | 1 (not suppressed) |
| `credentials are persisted` | substring of the message text | 1 (not suppressed) |

So the match target is the rule name. A user who follows the help text and writes a message fragment gets no suppression and no diagnostic.

## Why it matters

Suppression is the only way to silence a rule (there is no config or annotation path for most rules), so the help text is the primary documentation for it. A wrong target here means users either think suppression is broken, or write patterns broad enough to hit rules they did not intend.

## Suggested fix

Either change the help text to say the expression is matched against the rule name, or make the implementation match against the rendered message as documented. The first is the smaller change and matches how `-ignore` is used in practice.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)
